src/Settings/Settings.php

```diff
@@ -95,32 +76,25 @@ public function getIcon(): string
 
     /**
      * @throws SettingsException
-     *
-     * @return string
      */
     public function getFormClass(): string
     {
         $className = $this->metadata->getClass('form');
-        if (!\in_array(AbstractSettingsType::class, class_parents($className), true)) {
+        $parentClassNames = (array) class_parents($className) ?? [];
+        if (!\in_array(AbstractSettingsType::class, $parentClassNames, true)) {
             throw new SettingsException(sprintf('Class %s should extend %s', $className, AbstractSettingsType::class));
         }
 
         return $className;
     }
 
-    /**
-     * @param string $channelIdentifier
-     * @param string $localeIdentifier
-     * @param bool $withDefault
-     *
-     * @return array|null
-     */
     private function getCachedSettingsByChannelAndLocale(string $channelIdentifier, string $localeIdentifier, bool $withDefault): ?array
     {
         // With default?
         $varName = $withDefault ? 'settingsByChannelAndLocaleWithDefault' : 'settingsByChannelAndLocale';
+        /** @phpstan-ignore-next-line */
         if (!isset($this->{$varName}[$channelIdentifier])) {
-            $this->{$varName}[$channelIdentifier] = [];
+            $this->{$varName}[$channelIdentifier] = []; /** @phpstan-ignore-line */
```

Comment:
PHP Stan comment at the end of the line is what you want ?

Reply:
with class properties, this instruction isn't necessary